doc/architectural/CBMC-developer-guide.md

```diff
@@ -1,11 +1,196 @@
 \ingroup module_hidden 
-\page cprover-architecture-overview CProver Architecture Overview
+\ingroup module_hidden 
+\page cbmc-developer-guide CBMC Developer Guide
+
+\author César Rodríguez, Owen Jones
+
+\section section-compilation-and-development Compilation and Development
+
+## Makefiles ##
+
+First off, read the \ref cbmc-user-manual "CBMC User Manual". It describes
+how to get, build and use CBMC. This document covers the
+internals of the system and how to get started on development.
+
+## CMake files ##
+
+FIXME
+
+## Personal configuration: config.inc, macro DEBUG ##
+
+FIXME
+
+## Generating doxygen documentation ##
+
+Run `doxygen` in `/src`. HTML output will be created in `/doc/html`. The
+index page is `/doc/html/index.html`.
+
+Apart from the (user-orientated) CBMC user manual and this document, most
+of the rest of the documentation is inline in the code as `doxygen` and
+some comments. A man page for CBMC, goto-cc and goto-instrument is
+contained in the `doc/` directory and gives some options for these
+tools. All of these could be improved and patches are very welcome. In
+some cases the algorithms used are described in the relevant papers.
+
+## Coding standards ##
+
+See <a
+href="https://github.com/diffblue/cbmc/blob/master/CODING_STANDARD.md">
+`CODING_STANDARD.md`</a> file in the root of the CBMC repository.
+
+CPROVER is written in a fairly minimalist subset of C++; templates and
+meta-programming are avoided except where necessary.
+External library dependencies are avoided (only STL and a SAT solver
+are required). Boost is not used. The `util` directory contains many
```

Comment:
Where does this text come from?  It reminds me of some of my usual grammatical mistakes.

Reply:
Yes, I believe it is by you. It was in `cprover-archicture-overview.md`, which git has decided was renamed to `CBMC-developer-guide.md` in this PR. I think I moved it into a different section to match the order I was planning on.